### ISSUE TITLE
[Assistive tech and device support] Alert and status messages aren't announced without receiving focus. (11.22.1)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#63090]
#### VFS team name
Veteran Facing Forms

#### VFS product name
21P-0847

#### Point of Contact/Reviewers
Brian DeConinck - @briandeconinck - Accessibility

*For more information on how to interpret this ticket, please refer to the [Anatomy of a Staging Review issue ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Anatomy-of-a-Staging-Review-Issue-ticket.2060320997.html) guidance on Platform Website.

---

### Platform Issue
Alert and status messages aren't announced without receiving focus.

### Issue Details
When using the new memorable date component (eg. [/deceased-claimant-personal-information](https://staging.va.gov/supporting-forms-for-claims/substitute-claimant-form-21P-0847/deceased-claimant-personal-information)), the day and year inputs have hint text beneath the input indicating the maximum number of characters for each field. The hint text isn't announced when I test with a screen reader.

### Link, screenshot or steps to recreate
Steps to reproduce:
1. Navigate through the form flow to a page using the memorable date component, eg.  [/deceased-claimant-personal-information](https://staging.va.gov/supporting-forms-for-claims/substitute-claimant-form-21P-0847/deceased-claimant-personal-information).
2. Activate a screen reader for testing (I used NVDA on Windows).
3. Tab to the day input.
4. Type in a number.
5. Note that although "Max. 2 characters" appears on screen, the screen reader does not announce that text.

### VA.gov Experience Standard
[Category Number 11, Issue Number 22](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Other References
WCAG SC 4.1.3_AA

### Platform Recommendation
I think the implementation here is almost perfect, with one small issue. The max character text appears in a span with `aria-live="polite"`, which tells screen readers to watch that span for any changes and to announce the change. But the element doesn't exist on page load, it's added as you type your response. Screen readers are only watching for *changes to the contents of the element*, and the addition of the span itself doesn't trigger anything --- even with the `aria-live` attribute.

Recommended fix is for the span to be present but empty on page load, then the text gets added dynamically. As an `aria-live` region that should trigger screen readers to read out the max character text. 

Alternatively, is there any reason why that text isn't just there on page load? That could save some JS and avoid any ARIA issues entirely.

---

### VFS Guidance

- Close the ticket when the issue has been resolved or validated by your Product Owner
- If your team has additional questions or needs Platform help validating the issue, please comment on the ticket
- Some feedback provided may be out of scope for your iteration of the product, however, Platform's OCTO leadership has stated that all identified issues need to be documented and it is still your responsibility to resolve the issue.
- If you do not believe that this Staging Review issue ticket is the responsibility of your team, comment below providing an explanation and who you believe is responsible. Please tag the Point of Contact/Reviewers. Governance team will research and will follow up.

**Comment by jeana-adhoc** [comment]
This seems related to this issue in the [Design System](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1965). Will monitor.

**Comment by jeana-adhoc** [comment]
Screen reader now announces the date max-length.

Closing!
